Tests/RegoTests/BuiltinTests/NumbersTests.swift

```diff
@@ -0,0 +1,160 @@
+            expected: .success([1])
+        ),
+        BuiltinTests.TestCase(
+            description: "when a < b",
+            name: "numbers.range_step",
+            args: [1, 10, 3],
+            expected: .success([1, 4, 7, 10])
+        ),
+        BuiltinTests.TestCase(
+            description: "when a < b with large step",
+            name: "numbers.range_step",
+            args: [1, 10, 200],
+            expected: .success([1])
+        ),
+        BuiltinTests.TestCase(
+            description: "when a > b",
+            name: "numbers.range_step",
+            args: [4, -4, 2],
+            expected: .success([4, 2, 0, -2, -4])
+        ),
+        BuiltinTests.TestCase(
+            description: "when a > b with large step",
+            name: "numbers.range_step",
+            args: [2, 0, 100],
+            expected: .success([2])
+        ),
+        BuiltinTests.TestCase(
+            description: "with a is not an integer",
+            name: "numbers.range_step",
+            args: [1.2, 3, 1],
+            expected: .failure(
+                BuiltinError.evalError(
+                    msg: "operand 1 must be integer number but got floating-point number"))
+        ),
+        BuiltinTests.TestCase(
+            description: "when b is not an integer",
+            name: "numbers.range_step",
+            args: [1, 3.14, 1],
+            expected: .failure(
+                BuiltinError.evalError(
+                    msg: "operand 2 must be integer number but got floating-point number"))
+        ),
+        BuiltinTests.TestCase(
+            description: "when step < 0",
+            name: "numbers.range_step",
+            args: [3, 1, -1],
+            expected: .failure(
+                BuiltinError.evalError(
+                    msg: "step must be a positive integer"))
+        ),
+        BuiltinTests.TestCase(
+            description: "when step is not an integer",
+            name: "numbers.range_step",
+            args: [1, 3, 1.5],
+            expected: .failure(
+                BuiltinError.evalError(
+                    msg: "step must be integer number but got floating-point number"))
+        ),
+        BuiltinTests.TestCase(
+            description: "when a, b and step are floats with integer value",
+            name: "numbers.range_step",
+            args: [1.0000, 10.000, 3.00],
+            expected: .success([1, 4, 7, 10])
+        ),
+    ]
+
+    static var allTests: [BuiltinTests.TestCase] {
+        [
+            BuiltinTests.generateFailureTests(
+                builtinName: "numbers.range", sampleArgs: [1, 1], argIndex: 0, argName: "a",
+                allowedArgTypes: ["number"],
+                generateNumberOfArgsTest: true),
+            BuiltinTests.generateFailureTests(
+                builtinName: "numbers.range", sampleArgs: [1, 1], argIndex: 1, argName: "b",
+                allowedArgTypes: ["number"],
+                generateNumberOfArgsTest: false),
```

Comment:
💭 So I guess these cover the number of args in some way, and since we're doing this with the first arg (above, and in the argName="a" case below), that's sufficient. Sounds fair to me.

Reply:
Yes, there are similar calls below each arg....